File: Practica 4/python/FiguraGeometrica.py
import math

class FiguraGeometrica:

    def area(self, figura, *args):
        if figura == 'H':
            return (3 * math.sqrt(3) * args[0]**2) / 2
        elif figura == 'T':
            return ((args[0] + args[1]) * args[2]) / 2
        elif figura == 'I':
            return (args[0] * args[1]) / 2
        elif figura == 'R':
            return args[0] * args[1]
        elif figura == 'C':
            return math.pi * args[0]**2
        return -1

figura = FiguraGeometrica()

print("Área del hexágono:", figura.area('H', 5.0))
print("Área del trapecio:", figura.area('T', 10.0, 6.0, 5.0))
print("Área del triángulo:", figura.area('I', 4.0, 3.0))
print("Área del rectángulo:", figura.area('R', 6, 4))
print("Área del círculo:", figura.area('C', 3.0))

File: Practica 4/Java/FiguraGeometrica.java
public class FiguraGeometrica {
    public double area(double lado) {
        return (3 * Math.sqrt(3) * Math.pow(lado, 2)) / 2;
    }
    public double area(double baseMayor, double baseMenor, double altura) {
        return ((baseMayor + baseMenor) * altura) / 2;
    }
    public double area(double base, double altura) {
        return (base * altura) / 2;
    }
    public double area(int base, int altura) {
        return base * altura;
    }
    public double area(double radio, boolean c) {
        return Math.PI * Math.pow(radio, 2);
    }
    public static void main(String[] args) {
        FiguraGeometrica a=new FiguraGeometrica();
        System.out.println("Área del hexágono: " + a.area(5.0));
        System.out.println("Área del trapecio: " + a.area(10.0, 6.0, 5.0));
        System.out.println("Área del triángulo: " + a.area(4.0, 3.0));
        System.out.println("Área del rectángulo: " + a.area(6, 4));
        System.out.println("Área del círculo: " + a.area(3.0,true));
    }
}